Source/Core/Inc/UnFile.h

```diff
@@ -24,6 +24,39 @@ CORE_API extern DWORD GCRCTable[];
 	#define FIRST_BITFIELD   (0x80000000)
 #endif
 
+#if __INTEL__
+	#define INTEL_ORDER16(x)   (x)
+	#define INTEL_ORDER32(x)   (x)
+#else
+	// These macros are not safe to use unless data is UNSIGNED!
+	#define INTEL_ORDER16_unsigned(x)   ((((x)>>8)&0xff)+ (((x)<<8)&0xff00))
```

Comment:
You can use `__builtin_bswap16/32` for GCC/Clang. I don't know what other compilers you're planning to target, though.

Reply:
I'm using GCC, but since this is not a platform specific header I didn't think of putting builtins there.